src/gpu/SDL_gpu.c

```diff
@@ -41,6 +41,15 @@
     }
 
 #define CHECK_ANY_PASS_IN_PROGRESS                                              \
+    if (                                                                        \
+        ((CommandBufferCommonHeader *)commandBuffer)->renderPass.inProgress ||  \
+        ((CommandBufferCommonHeader *)commandBuffer)->computePass.inProgress || \
+        ((CommandBufferCommonHeader *)commandBuffer)->copyPass.inProgress) {    \
+        SDL_assert_release(!"Pass already in progress!");                       \
+        return;                                                            \
+    }
+
+#define CHECK_ANY_PASS_IN_PROGRESS_RETURN_NULL                                              \
```

Comment:
Maybe we should fuse these two via `#define CHECK_ANY_PASS_IN_PROGRESS(retval)`? There are some other macros that already do this.

Reply:
I'm not sure how to pass an empty arg, MSVC complains about it. We're already doing this for the command buffer check so I think it's fine.